docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+
+- The `@UdfParameter` annotation lets you give the function parameters names. These are rendered when using the `DESCRIBE` statement on a function.
+
+- This UDF uses an external parameter, `ksql.functions.formula.base.value`. When a UDF implements the `Configurable` interface, it will be invoked once as the server starts up. `configure()` supplies a map of the parameters that ksqlDB server was started with. You will see how this value is populated later in the guide.
+
+!!! warning
+    External parameters do not yet work for tabular or aggregation functions.
+    Support for these function types will be added soon.
+
+Either continue following this guide by implementing more functions or skip ahead to [compiling the classes](#add-the-uberjar-to-the-classpath) so you can use the functions in ksqlDB.
+
+### Tabular functions
+
+A tabular function (UDTF for short) takes one row as input and produces zero or more rows as output. This is sometimes called "flat map" or "mapcat" in different programming languages. Use this when a value represents many smaller values and needs to be "exploded" into its individual parts to be useful.
+
+Create a file at `src/main/java/com/example/IndexSequenceUdtf.java` and populate it with the following code. This UDTF takes one parameter as input, an array of any type, and returns a sequence of rows, where each element is the element in the array concatenated with its index position as a string.
+
+```java
+package my.example;
+
+import io.confluent.ksql.function.udtf.Udtf;
+import io.confluent.ksql.function.udtf.UdtfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@UdtfDescription(name = "index_seq",
+                 author = "example user",
+                 version = "1.5.0",
+                 description = "Disassembles a sequence and produces new elements concatenated with indices.")
+public class IndexSequenceUdtf {
+
+    private final String DELIMITER = "-";
+
+    @Udtf(description = "Takes an array of any type and returns rows with each element paired to its index.")
+    public <E> List<String> indexSequence(@UdfParameter(value = "s") List<E> x) {
+        List<String> result = new ArrayList<>();
+
+        for(int i = 0; i < x.size(); i++) { 
+            result.add(x.get(i) + DELIMITER + i);
+        }
+
+        return result;
+    }
+
+}
+```
+
+Notice how:
+
+- The UDTF returns a Java `List`. This is the collection type that ksqlDB expects all tabular functions to return.
+
+- This UDTF uses Java generics which allow it operate over any [ksqlDB supported types](../../concepts/functions/#supported-types). Use this if you want to express logic that operates uniformly over many different column types. The generic parameter must be declared at the head of the method since you can have multiple signatures, each with a different generic type parameter.
```

Comment:
```suggestion
- This UDTF uses Java generics, which enable operating over any [ksqlDB supported types](../../concepts/functions/#supported-types). Use this if you want to express logic that operates uniformly over many different column types. The generic parameter must be declared at the head of the method, since you can have multiple signatures, each with a different generic type parameter.
```